File: 2016/algorithm/Nim Game/nimGame-20160505.cc
#include <iostream>
#include <string>
#include <cstring>


using namespace std;

class Solution {
public:
    bool canWinNim(int n) {
        return ! (n % 4 == 0);
    }
};


int main() {
    Solution sol;
    string concern;
    cout << "Let's see if you can win!\n"
        << "Hit 'q' to quit!\n";

    while(cin >> concern) {
        if(concern != "q") {
            int real = stoi(concern, nullptr, 0);
            if (sol.canWinNim(real)) {
                cout << "Can Win Nim!\n";
            }
            else {
                cout << "Oops! Can't win!\n";
            }
        }
        else
            break;
    }

    

    return 0;
}

File: 2016/algorithm/292_Nim_Game/nimGame-20160505.py
class Solution(object):
    def canWinNim(self, n):
        return not(n % 4 is 0)


sol = Solution()
concern = "Hit \'q\' to quit!\n"
while(True):
    concern = raw_input("")
    if(concern is not 'q'):
        real = int(concern)
        if(sol.canWinNim(real)):
            print 'Can Win!\n'
        else:
            print 'Oops! Will definitely lose!\n'
    else:
        break;
